modules/administration-guide/attachments/restricted-environment/prepare-restricted-environment.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+#
+# Copyright (c) 2022 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+
+# Fail on error
+set -e
+
+while [ $# -gt 0 ]; do
+   if [[ $1 == *"--"* ]]; then
+        param="${1/--/}"
+        declare $param="$2"
+        # echo $1 $2 // Optional to see the parameter:value result
+   fi
+  shift
+done
+
+# OpenShift cluster version
+export ocp_ver="${ocp_ver:?Define the variable}"
+
+# Operator catalogs
+export devworkspace_operator_bundle="${devworkspace_operator_bundle:?Define the variable}"
+export prod_operator_bundle="${prod_operator_bundle:?Define the variable}"
+
+# Operators
+export devworkspace_operator="devworkspace"
+export prod_operator="${prod_operator:?Define the variable}"
+
+# Destination registry
+export my_registry="${my_registry:-default-route-openshift-image-registry.apps-crc.testing}"
+export my_catalog="${my_catalog:-restricted-environment-install}"
+export my_operator_index="$my_registry/$my_catalog/my-operator-index:v${ocp_ver}"
+
+# Create local directories
+mkdir -p "${my_catalog}/${devworkspace_operator}"
+mkdir -p "${my_catalog}/${prod_operator}"
+
+# Get metadata for the operator catalog channels and packages.
+opm render "${devworkspace_operator_bundle}" > "${my_catalog}/${devworkspace_operator}/render.json"
+
+opm render "${prod_operator_bundle}" > render.json
+jq "select(.name == \"${prod_operator}\" or .package == \"${prod_operator}\")" render.json > "${my_catalog}/${prod_operator}/render.json"
```

Comment:
With current filtering, we still have *all* che packages. Meaning mirroring is very long. Maybe we could filter out all old packages?